### PR TITLE
feat(aria-allowed-attr): report violation for non-global ARIA attrs on element without role

### PR DESCRIPTION
**Pull request by straker**
This change reintroduces the change to [report violations on elements without a role for non-global ARIA attrs](https://github.com/dequelabs/axe-core/pull/3102), which was [reverted](https://github.com/dequelabs/axe-core/pull/3243) when we accidentally released it.

This also introduces a fix for custom elements where we will report Needs Review for any custom element that uses a non-global ARIA attribute (rather than a failure).

Closes issue: https://github.com/dequelabs/axe-core/issues/3241

### CODE REVIEW

**Comment by straker** [comment]
The error from aria practices seems to be a legit issue on aria practices? The violation reported on the `button_idl.html` page is from an anchor element which uses `aria-pressed`, which shouldn't be allowed.

This is the page: https://www.w3.org/TR/wai-aria-practices-1.2/examples/button/button_idl.html

And the error:

```js
{
    "any": [
        {
            "id": "aria-allowed-attr",
            "data": [
                "aria-pressed=\"false\""
            ],
            "relatedNodes": [],
            "impact": "critical",
            "message": "ARIA attribute is not allowed: aria-pressed=\"false\""
        }
    ],
    "all": [],
    "none": [],
    "impact": "critical",
    "html": "<a tabindex=\"0\" id=\"toggle\" aria-pressed=\"false\">\n          Mute\n          <svg aria-hidden=\"true\" focusable=\"false\">\n            <use xlink:href=\"#icon-sound\"></use>\n          </svg>\n        </a>",
    "target": [
        "#toggle"
    ],
    "failureSummary": "Fix any of the following:\n  ARIA attribute is not allowed: aria-pressed=\"false\""
}
```

File: lib/checks/aria/aria-allowed-attr.json [hunk 1]
```diff
@@ -16,7 +16,8 @@
       "fail": {
         "singular": "ARIA attribute is not allowed: ${data.values}",
         "plural": "ARIA attributes are not allowed: ${data.values}"
-      }
+      },
+      "incomplete": "ARIA attribute is not allowed on custom element: ${data.values}"
```
Comment (WilcoFiers) [inline lib/checks/aria/aria-allowed-attr.json:20]: How about:

```suggestion
      "incomplete": "Check that there is no problem the ARIA attribute is ignored on this element: ${data.values}"
```

**Review by WilcoFiers — CHANGES_REQUESTED** [review]

File: lib/checks/aria/aria-allowed-attr-evaluate.js [hunk 2]
```diff
@@ -82,7 +82,7 @@ function ariaAllowedAttrEvaluate(node, options, virtualNode) {
 
   if (invalid.length) {
     this.data(invalid);
-    return false;
+    return isHtmlElement(virtualNode) ? false : undefined;
```
Comment (WilcoFiers) [inline lib/checks/aria/aria-allowed-attr-evaluate.js:85]: I think this exception should only be used:
1. If the element is not an HTML element
2. If the element is not focusable
3. If the element has no role

File: lib/checks/aria/aria-allowed-attr.json [hunk 3]
```diff
@@ -17,7 +17,7 @@
         "singular": "ARIA attribute is not allowed: ${data.values}",
         "plural": "ARIA attributes are not allowed: ${data.values}"
       },
-      "incomplete": "ARIA attribute is not allowed on custom element: ${data.values}"
+      "incomplete": "Check that there is no problem the ARIA attribute is ignored on this element: ${data.values}"
```
Comment (WilcoFiers) [inline lib/checks/aria/aria-allowed-attr.json:20]:
```suggestion
      "incomplete": "Check that there is no problem if the ARIA attribute is ignored on this element: ${data.values}"
```

**Review by WilcoFiers — APPROVED** [review]

**Pull request merged**